### PR TITLE
[FLINK-3077][cli] Add version option to Cli.

### PR DESCRIPTION
**Pull request by sachingoel0101**
Adds a version option (-v, --version) to the Cli, which prints the implementation version and the git commit for the build.

### CODE REVIEW

**Comment by sachingoel0101** [comment]
Also, resolves FLINK-3056.
capture10

**Comment by sachingoel0101** [comment]
Also resolves FLINK-3023.
capture10

File: flink-core/src/main/java/org/apache/flink/util/VersionUtils.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				//
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Returns the commit id of the source from which the flink jar is built.
+	 */
+	public static String getCommitId() {
```
Comment (rmetzger) [inline flink-core/src/main/java/org/apache/flink/util/VersionUtils.java]: Afaik this is a re-implementation of `EnvironmentInformation.getRevisionInformation()`. Also the "getFlinkVersion" method is already implemented in Flink.
Can you re-use the method from the EnvironmentInformation for the CLI frontend and the web interface?

File: flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java [hunk 2]
```diff
@@ -1000,11 +1001,17 @@ else if (action.equals("-h") || action.equals("--help")) {
...
+		else if (action.equals("-v") || action.equals("--version")) {
+			System.out.println("Version: " + VersionUtils.getFlinkVersion() + ", Commit ID: " + VersionUtils.getCommitId());
+			return 0;
+		}
 		else {
 			System.out.printf("\"%s\" is not a valid action.\n", action);
 			System.out.println();
 			System.out.println("Valid actions are \"run\", \"list\", \"info\", or \"cancel\".");
 			System.out.println();
+			System.out.println("Specify the version option (-v or --version) to print flink version.");
```
Comment (rmetzger) [inline flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java]: Can you write "Flink" uppercase here?

File: flink-core/src/main/java/org/apache/flink/util/VersionUtils.java (hunk 1 above)
Comment (sachingoel0101) [inline flink-core/src/main/java/org/apache/flink/util/VersionUtils.java]: Ah. I was entirely unaware of this class. 
There's also a method already for accessing the revision id, albeit abbreviated. Is the short rev id okay or should I just place a full rev id string in the same class?

Comment (rmetzger) [inline flink-core/src/main/java/org/apache/flink/util/VersionUtils.java]: The short rev is okay.

**Comment by sachingoel0101** [comment]
@rmetzger I have split this PR to separate the Web dashboard specific work into another PR #1422

**Comment by StephanEwen** [comment]
Does it show the commit only for snapshot versions?

**Comment by rmetzger** [comment]
From the code it seems that its always showing the commit id.
@sachingoel0101: I guess Stephan is asking for this because the the old web interface was only showing the commit id for `-SNAPSHOT` versions. For releases, the commit id is actually irrelevant.
Can you change the behavior also in the web interface PR?

**Comment by StephanEwen** [comment]
I think for release builds, the commit ID may be null, depending on how the uploaded release binary was built...

**Comment by rmetzger** [comment]
Oh, you are right. We are building the binary releases from a directory without git, so the git version plugin in maven will not put any commit ID.

**Comment by sachingoel0101** [comment]
Do I need to address this? Perhaps we can check if the commit id is null and not print it in such  a case?

**Comment by StephanEwen** [comment]
I think this code here should print the commit ID only when it is non null...

**Comment by mxm** [comment]
+1 please only print if the version is not null.

**Comment by sachingoel0101** [comment]
Makes sense. I will address it as per Stephan's comments.
I won't be able to get to it till this weekend though as I'm on vacation.

**Comment by sachingoel0101** [comment]
I have made the required changes. @StephanEwen can you take a look again?

**Comment by sachingoel0101** [comment]
Ping.
Also, what is the procedure for creating a release artifact locally[without GPG keys, Apache id, etc.] so I can check that this works okay for those too?

**Comment by mxm** [comment]
>Also, what is the procedure for creating a release artifact locally[without GPG keys, Apache id, etc.] so I can check that this works okay for those too?

The release scripts do nothing fancy. Essentially, they just set the release version and delete the .git folder. You can do the same locally. The GPG keys and Apache id is just for actual deployment of the build artifacts.

**Comment by rmetzger** [comment]
I'll merge this change ...

**Pull request closed without merging**